xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -730,8 +721,7 @@ private void handleLdsResponseForListener(DiscoveryResponse ldsResponse) {
       listenerWatcher.onListenerChanged(listenerUpdate);
     } else {
       if (ldsRespTimer == null) {
-        listenerWatcher.onError(Status.NOT_FOUND.withDescription("did not find listener for "
-            + listenerPort));
+        listenerWatcher.onResourceDoesNotExist(":" + listenerPort);
```

Comment:
The actual resource name used in the request in this case was null/empty string. But you are reporting it as `":" + listenerPort`. I don't have a better suggestion other than to put a comment mentioning that.

Reply:
Yeah, I noticed that. Maybe the doc on `onResourceDoesNotExist()` is somewhat imprecise. The argument is the resource name that we are looking for but does not exist. I am going to update that.

For Listener watcher, this is kind of ambiguous as it sits on the border of the design (i.e., the way we watch Listener is to request all and select the one we are seeking for). But at least, using `":" + listenerPort` turns out to be better than using null/empty string.

Actually `ConfigWatcher` also has somewhat similar issue, since resource-not-found may be caused by the intermediate RDS lookup, which uses a different resource name (other than what was used for registering the watcher) for the discovery request. In that case, the `onResourceDoesNotExist()` callback is invoked (and IMO it should be) with the actual requested RDS resource name.